modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/db/Statement.scala

```diff
@@ -0,0 +1,229 @@
+  }
+  case class SchemaExists(schema: String) extends Statement {
+    def toFragment: Fragment =
+      sql"SELECT schema_name FROM information_schema.schemata WHERE schema_name = $schema"
+  }
+  case class GetVersion(schema: String, tableName: String) extends Statement {
+    def toFragment: Fragment =
+      sql"""
+           SELECT obj_description(oid)::TEXT
+            FROM pg_class
+            WHERE relnamespace = (
+               SELECT oid
+               FROM pg_catalog.pg_namespace
+               WHERE nspname = $schema)
+            AND relname = $tableName
+      """.stripMargin
+  }
+
+
+  case class SetSchema(schema: String) extends Statement {
+    def toFragment: Fragment =
+      Fragment.const0(s"SET search_path TO $schema")
+  }
+  case class GetColumns(tableName: String) extends Statement {
+    def toFragment: Fragment =
+      sql"""SELECT "column" FROM PG_TABLE_DEF WHERE tablename = $tableName"""
+  }
+
+  // Manifest
+  case class ManifestAdd(schema: String, message: LoaderMessage.ShreddingComplete) extends Statement {
+    def toFragment: Fragment = {
+      val tableName = Fragment.const(s"$schema.manifest")
+      val types = message.types.asJson.noSpaces
+      // Redshift JDBC doesn't accept java.time.Instant
+      sql"""INSERT INTO $tableName
+        (base, types, shredding_started, shredding_completed,
+        min_collector_tstamp, max_collector_tstamp, ingestion_tstamp,
+        compression, processor_artifact, processor_version, count_good)
+        VALUES (${message.base}, $types,
+        ${Timestamp.from(message.timestamps.jobStarted)}, ${Timestamp.from(message.timestamps.jobCompleted)},
+        ${message.timestamps.min.map(Timestamp.from)}, ${message.timestamps.max.map(Timestamp.from)},
+        getdate(),
+        ${message.compression.asString}, ${message.processor.artifact}, ${message.processor.version}, ${message.count})"""
+    }
+  }
+  case class ManifestGet(schema: String, base: S3.Folder) extends Statement {
+    // Order of columns must remain the same to conform Entity properties
+    def toFragment: Fragment =
+      sql"""SELECT ingestion_tstamp,
+                   base, types, shredding_started, shredding_completed,
+                   min_collector_tstamp, max_collector_tstamp,
+                   compression, processor_artifact, processor_version, count_good
+            FROM ${Fragment.const0(schema)}.manifest WHERE base = $base"""
+  }
+
+  // Schema DDL
+  case class CreateTable(ddl: redshift.CreateTable) extends Statement {
+    def toFragment: Fragment =
+      Fragment.const0(ddl.toDdl)
+  }
+  case class CommentOn(ddl: redshift.CommentOn) extends Statement {
+    def toFragment: Fragment =
+      Fragment.const0(ddl.toDdl)
+  }
+  case class DdlFile(ddl: redshift.generators.DdlFile) extends Statement {
+    def toFragment: Fragment = {
+      val str = ddl.render.split("\n").filterNot(l => l.startsWith("--") || l.isBlank).mkString("\n")
```

Comment:
`l.startsWith("--")` - can a ddl file also legitimately start with white space before the `--`?

Reply:
Good point. It can. But in this case we also have a full control over a produced DDL - I don't think we should be handling all corner cases like that. And in fact, this is just to make the debugging easier, there's nothing wrong with having those comments.